index.android.js

```diff
@@ -30,12 +30,18 @@ componentWillMount: function(){
   var self = this;
   FBLoginManager.getCurrentToken(function(token){
     if(itypeof(token) === 'string' && token.length > 0){
-      self.setState({isLoggedIn:true, buttonText: self.state.statics.logoutText});
+      self.setState({isLoggedIn:true, buttonText: self._getLoginText()});
     }else{
-      self.setState({isLoggedIn:false, buttonText: self.state.statics.loginText});
+      self.setState({isLoggedIn:false, buttonText: self._getLogoutText()});
     }
   })
 },
+  _getLoginText: function(){
+      return this.props.loginText || this.state.statics.loginText;
```

Comment:
Did you test this to ensure the right text is showing in the respective States. Because if isloggedin is set to true, I would assume you would want to show direction for logging out